### ISSUE TITLE
"Hide results on init" example breaks URL sync

### ISSUE THREAD

**Issue by Spone** [algolia/instantsearch.js#2173]
**Do you want to request a *feature* or report a *bug*?**

A bug

**Bug: What is the current behavior?**

I use the example code from the documentation: https://community.algolia.com/instantsearch.js/documentation/#hide-results-on-init

```js
var search = instantsearch({
  [...],
  searchFunction: function(helper) {
    var searchResults = $('.search-results');
    if (helper.state.query === '') {
      searchResults.hide();
      return;
    }
    helper.search();
    searchResults.show();
  }
}
```

Here is a JSFiddle: https://jsfiddle.net/4jqgawpk/3/

Open the HTML iframe: https://fiddle.jshell.net/4jqgawpk/3/show/ (please try in Firefox, somehow the URL is not updated in Chrome, probably because of JSFiddle)

Type a search, wait for the URL params to update

Empty the search box

The URL params are not reset, the last search query is still there.

**Bug: What is the expected behavior?**

When I empty the search box, I expect the URL to be "clean" (at least the `q` param should be empty).

**Bug: What browsers are impacted? Which versions?**

Firefox latest, Chrome latest

**Bug: What is the proposed solution?**

There should probably be a way to use the JS helper to reset the URL params. If possible, the example from the documentation should be updated to use it.

**Comment by bobylito** [comment]
Hi @Spone 

I came up with a solution to your problem. Indeed now InstantSearch.js relies on the `search` event to update the URL, and in your code you are preventing the search from happening. But there is a way to do that in the [following jsFiddle](https://jsfiddle.net/bobylito/e2jd75oL/1/).

What I did is change the implementation of the searchFunction to have this:

```js
  searchFunction: function(helper) {
    var searchResults = $('#movies');
    if (helper.state.query === '') {
      movies.helper.once('result', function() {
        searchResults.hide();
      })
    }
    helper.search();
    searchResults.show();
  }
```

In this case, there will be a search but as soon as the results come back it hides the content of the hits.

**Comment by Spone** [comment]
Thanks @bobylito for this solution!

It's not ideal as it triggers a search every time the page loads, even if the user is not searching (I'm trying to limit the number of requests to Algolia to stay in our quota).
In my case, the items on the page are already rendered server-side.

Do you see another option?

**Comment by bobylito** [comment]
Not for now. I'll be looking for a fix :/

**Comment by bobylito** [comment]
This bug has its roots in the attempt to fix #2105 Revisiting the fix to provide a less invasive one.
